shiftcommerce/shift-react-components
# Part 1: Checkout integration -Payment Method Page Components

## Overview

This PR adds components and styling used on the Payment Method checkout page.

It also moves the checkout cart buttons into their own component `CheckoutCartButtons`. When `continueButtonProps` are not passed, the `Continue Shopping` button is rendered at full-width

![Payment Method Page](https://user-images.githubusercontent.com/4486874/54922676-8dda5080-4f00-11e9-9450-1eb697c4e13c.png)

Related PRs:
[Shift Next PR](https://github.com/shiftcommerce/shift-next/pull/10)
[Shift Front End React](https://github.com/shiftcommerce/shift-front-end-react/pull/846)


#### * Known Issues to be address in subsequent PR include: 
- PayPal script is currently being rendered on all checkout pages. Ideally we need to only have it in checkout 1
- On the Payment Methods page, if you scroll up the PayPal button disappears. This is because the page re-renders and we suspect its to do with the way the `CustomHead` component is being included
- When you are redirected from PayPal, we are getting this error in the console -

```
unhandled_error {err: "Error: Window closed↵    at https://www.paypal.com…mex,visa,mastercard,discover,jcb,elo,hiper:1:4201", timestamp: "1553511012713", referer: "localhost:3000", uid: "cc1340311c_mta6ndq6mze", env: "sandbox"}
```

## QA

1. checkout branch `issue/790_paypal_checkout_integration_part_1`

2. In the `shift-front-end-react` project, checkout branch `issue/790_paypal_checkout_integration` and boot app (assumption is that you are using yalc)

3. Add items to cart and navigate to the first step of checkout - `payment-method` selection

4. Check that the `Continue Shopping` btn in the cart should be full-width

5. Check that the `Payment Method` step is present and highlighted

6. Check that there are payment method options displayed on the page

7. Check that  clicking the `Pay By Credit/Debit Card` button results in the page navigating to checkout step 2 - `Shipping Address`

8. Check that clicking the `PayPal` button in the page navigates to the PayPal website. 

![Payment Method Selection](https://user-images.githubusercontent.com/4486874/54922925-0fca7980-4f01-11e9-8ac0-28a121202637.png)

9. Complete all steps on PayPal and click the continue button

![Shipping Address Selection](https://user-images.githubusercontent.com/4486874/54922936-13f69700-4f01-11e9-9bb5-2b8159d5d604.png)

10. When you have been redirected back to the site, check that the shipping address summary is present and that you are redirect to checkout 3 - shipping method. Also ensure that the `Edit` address button is not present and that clicking the shipping address step in the checkout header steps, redirects you back to checkout 1 - Payment Method selection

![Checkout 3 - Shipping Methods](https://user-images.githubusercontent.com/4486874/54924891-2bd01a00-4f05-11e9-9541-e39d63a8b464.png)
